ICP-13858 - fix for setting the thermostat mode to OFF

KKlimczukS opened the pull request:
ICP-13858 - fix for setting the thermostat mode to OFF

@greens
Please, could You review it ?

cc: @SmartThingsCommunity/srpol-pe-team

KKlimczukS committed:
ICP-13858 - fix for setting the thermostat mode to OFF

greens reviewed (changes requested):
You'll have the same bug for fan mode as well so you might want to fix that as well. Also you control every place these functions are called so I don't think you need the null check at all, frankly.

KKlimczukS committed:
ICP-13858 - fixes setting the thermostat fan mode to ON

KKlimczukS committed:
ICP-13858 - removed null checks (since the methods are called with proper parameters)

KKlimczukS commented:
> You'll have the same bug for fan mode as well so you might want to fix that as well. Also you control every place these functions are called so I don't think you need the null check at all, frankly.

@greens Thanks for pointing that out. I fixed that one too (since methods are called with proper parameters - i removed the null check according to Your suggestion)
